control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -2071,6 +2077,50 @@ func (r *HostedControlPlaneReconciler) reconcileClusterNetworkOperator(ctx conte
 	return nil
 }
 
+// reconcileDNSOperator ensures that the management cluster has the expected DNS
+// operator deployment and kubeconfig secret for the hosted cluster.
+func (r *HostedControlPlaneReconciler) reconcileDNSOperator(ctx context.Context, hcp *hyperv1.HostedControlPlane, releaseImage *releaseinfo.ReleaseImage, createOrUpdate upsert.CreateOrUpdateFN) error {
```

Comment:
This will reconcile it in the mgmt cluster, not in the guest cluster. You would need to add this into the hosted cluster config operator (control-plane-operator/hostedclusterconfigoperator/controllers/resources/), alternatively you could use the pre-existing kubeconfig in the `service-network-admin-kubeconfig` secret just for the init container